types.nix

```diff
@@ -235,9 +236,80 @@ rec {
         type = types.attrsOf lvm_vg;
         default = {};
       };
+      nodev = mkOption {
+        type = types.attrsOf (diskoLib.subType { inherit tmpfs; });
+        default = {};
+      };
     };
   };
 
+  tmpfs = types.submodule ({ config, ... }: {
+    options = {
+      type = mkOption {
+        type = types.enum [ "tmpfs" ];
+        internal = true;
+      };
+      size = mkOption {
+        type = types.nullOr types.str; # TODO size type
+        default = null;
+      };
+      mountpoint = mkOption {
+        type = optionTypes.absolute-pathname;
+        default = config._module.args.name;
+      };
+      mountOptions = mkOption {
+        type = types.listOf types.str;
+        default = [];
+      };
+      _meta = mkOption {
+        internal = true;
+        readOnly = true;
+        type = diskoLib.jsonType;
+        default = {
+        };
+      };
+      _create = mkOption {
+        internal = true;
+        readOnly = true;
+        type = types.str;
+        default = "";
+      };
+      _mount = mkOption {
+        internal = true;
+        readOnly = true;
+        type = diskoLib.jsonType;
+        default = {
+          fs.${config.mountpoint} = ''
+            if ! findmnt tmpfs "/mnt${config.mountpoint}" > /dev/null 2>&1; then
+              mount -t tmpfs none "/mnt${config.mountpoint}" \
+              ${lib.optionalString (!isNull config.size) "-o size=${config.size}"} \
+              ${concatStringsSep " " config.mountOptions} \
+              -o X-mount.mkdir
+            fi
```

Comment:
I think all it would take would be to not have a `size` mountoptions and add an fs type instead of hardcoding `tmpfs`.